origin-dapp/src/components/web3-provider.js

```diff
@@ -433,30 +508,38 @@ class Web3Provider extends Component {
 
     // on account detection
     if (curr !== prev) {
-      // start over if changed
-      prev !== null && window.location.reload()
+      // TODO: fix this with some route magic!
```

Comment:
@crazybuster This breaks the expected behavior: to reload everything when the account changes. What are we trying to accomplish with the route conditions?